teacher.rb

```diff
@@ -0,0 +1,13 @@
+# Represents a teacher with specific attributes and
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(age, specialization, name, parent_permission)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_service?
+    true
+  end
+end
```

Comment:
- [ ] When I try to create an instance of the `Teacher` class, I get this error:

![image](https://github.com/mosidrum/OOP-school-library/assets/77106962/081aaa04-6b0b-412b-bfe6-2001963e8ee9)

---

The `parent_permission` class attribute is passed in as a _**keyword**_ `(parent_permission: true)` argument in the superclass `Person`. Therefore when you call the `super` method in the initialize method of the `Teacher` class, you need to specify the value that will be passed to the `parent_permission` attribute from the superclass. In this case, you want to pass the value of the `parent_permission` being passed as a parameter when creating an instance of the `Teacher` class through the `initialize` method to the `parent_permission` attribute when calling super. It is after this that conditions specified in the initialize method of the superclass `Person` will be attainable in the child class `Teacher`. Kindly take a look at the sample code below 👍 😎:

```ruby
def initialize(age, specialization, name, permission: true)
  super(age, name, parent_permission: permission)
  @specialization = specialization
end
```

Then you can create a new instance of the `Student` class by doing this:

```ruby
teacher = Teacher.new(34, 'Math', 'Jane Doe', parent_permission: true)
```